### PR TITLE
Patch user to ES signal

### PR DESCRIPTION
**Pull request by millerdev**
Tests that want to sync users to ES should use `@sync_users_to_es()` decorator/context manager.

Based on https://github.com/dimagi/commcare-hq/pull/27876

### CODE REVIEW

**Review by kaapstorm — APPROVED** [review]
> In the process of moving it to a new module I also devised a more reliable way to have the patch applied only when running tests (using coreqh.tests.noseplugins.patches) 

Awesome! I'm very glad you could find a way to apply the patch cleanly.

File: corehq/util/es/testing.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+from contextlib import contextmanager
+from mock import patch
+
+from django.conf import settings
+
+_sync_user_to_es_patch = patch('corehq.apps.users.signals.send_to_elasticsearch')
+
+
+def patch_es_user_signals():
+    assert settings.UNIT_TESTING
```
Comment (kaapstorm) [inline corehq/util/es/testing.py:10]: :+1:

File: corehq/tests/noseplugins/patches.py [hunk 2]
```diff
@@ -12,6 +14,7 @@ def options(self, parser, env):
     def begin(self):
         patch_assertItemsEqual()
         fix_freezegun_bugs()
+        patch_es_user_signals()
```
Comment (sravfeyn) [inline corehq/tests/noseplugins/patches.py:17]: Nice! 👍

**Review by sravfeyn — APPROVED** [review]

**Comment by sravfeyn** [comment]
Thanks for the very clean implementation @millerdev !

I kicked the build again as it looked unrelated.

**Review by mkangia — APPROVED** [review]
Clean 💯

**Comment by snopoke** [comment]
build failing on kafka timeout - merging master should fix it

**Comment by snopoke** [comment]
Should this be documented anywhere?

**Comment by millerdev** [comment]
> Should this be documented anywhere?

@snopoke I'll be happy to add documentation. Do you have any ideas of a good place for that?

**Comment by snopoke** [comment]
@millerdev I had thought we had some testing docs in the repo but it's very bare: https://commcare-hq.readthedocs.io/testing.html

Perhaps a new doc on how our testing is set up and run? No need to include a lot of detail, just a few pointers for where to look should be enough.

**Comment by millerdev** [comment]
I'll add docs in a follow-up PR.

**Pull request merged**